infrastructure/terraform.tf

```diff
@@ -19,7 +19,7 @@ terraform {
   required_providers {
     azurerm = {
       source  = "hashicorp/azurerm"
-      version = "~> 3.112"
+      version = "~> 4.0.0"
```

Comment:
previous configuration ensured that not only patch releases were pulled in automatically, but minor releases as well. this has worked reliably for about a year now. is there a reason to restrict automatic updates to patch only? If not, recommend to change to "~> 4.0"